test/integration/items_store.go

```diff
@@ -18,36 +18,37 @@ type (
 
 	// Store allows operations on items.
 	Store interface {
-		Create(i Item) (Item, error)
-		Find(query bson.M, next string, previous string, limit int, sortAscending bool, paginatedField string, collation mgo.Collation) ([]Item, mongocursorpagination.Cursor, error)
+		Create(i *Item) (*Item, error)
+		Find(query interface{}, next string, previous string, limit int, sortAscending bool, paginatedField string, collation mgo.Collation) ([]*Item, mongocursorpagination.Cursor, error)
 		EnsureIndices() error
 	}
 
-	mongoStore struct {
+	mgoStore struct {
 		col *mgo.Collection
 	}
 )
 
-// NewMongoStore returns a new Store.
-func NewMongoStore(col *mgo.Collection) Store {
-	return &mongoStore{
+// NewMgoStore returns a new Store that uses mgo.
+func NewMgoStore(col *mgo.Collection) Store {
+	return &mgoStore{
 		col: col,
 	}
 }
 
 // Create creates an item in the database and returns it
-func (m *mongoStore) Create(c Item) (Item, error) {
+func (m *mgoStore) Create(c *Item) (*Item, error) {
 	c.ID = bson.NewObjectId() // Generate ObjectID
 	c.CreatedAt = bson.Now().UTC()
 	return c, m.col.Insert(c)
 }
 
 // Find returns paginated items from the database matching the provided query
-func (m *mongoStore) Find(query bson.M, next string, previous string, limit int, sortAscending bool, paginatedField string, collation mgo.Collation) ([]Item, mongocursorpagination.Cursor, error) {
+func (m *mgoStore) Find(query interface{}, next string, previous string, limit int, sortAscending bool, paginatedField string, collation mgo.Collation) ([]*Item, mongocursorpagination.Cursor, error) {
```

Comment:
It's ok to only test the pointer case here if you add a unit test for the value case.

Reply:
There are only minor changes to the unit-tests, all but one still use values and not pointers.
The main difference is that the tests handle any `results` array and not a specific type since we're only looking at the length anyway!
See the comment below.